### PR TITLE
Twig Templates Manual

### PR DESCRIPTION
**Pull request by mlwebworker**
only german language

### CODE REVIEW

**Review by netzarbeiter — APPROVED** [review]
Thank you, Maren

File: docs/manual/layout/templates/twig/_index.en.md [hunk 1]
```diff
@@ -1,58 +1,55 @@
...
-## Getting started
 
-You can use Twig templates at any place you would use a Contao PHP template, just with a different file extension 
-(`.html.twig` instead of `.html5`). It’s even possible to extend Contao PHP templates from within your Twig templates.
+## Twig templates in Contao core
 
-Go ahead and place a `fe_page.html.twig` in your template directory - this example will add a friendly headline above the main section 
-and keep everything else the same:
+In Contao 5, Twig templates are provided for many core elements. This means that template adjustments must also be made
+in the Twig templates.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/_index.en.md]:
```suggestion
in Twig templates.    
```

File: docs/manual/layout/templates/twig/twig-syntax.en.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+section [Twig for template designers](https://twig.symfony.com/doc/3.x/templates.html) is recommended.
+{{% /notice %}}
+
+
+## Identifier
+
+In Twig, the following three identifiers are used.
+
+* `{# ... #}` - [Comments](#comments)
+* `{{ ... }}` - [Output of variable](#output-of-variables)
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-syntax.en.md]:
```suggestion
* `{{ ... }}` - [Output of a variable](#output-of-variables)
```

File: docs/manual/layout/templates/twig/twig-syntax.en.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+```twig
+<p>output: {{ name_of_variable }} </p>
+```
+{{% /example %}}
+
+
+### Commands and control structures
+
+In the broadest sense, this refers to everything that is connected with control when variables are output.
+is connected.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-syntax.en.md]:
```suggestion
In the broadest sense, this refers to everything that is connected to controlling the output of variables.
```

File: .idea/docs.iml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/deepl/src" isTestSource="false" packagePrefix="Contao\Docs\DeeplTranslator\" />
+      <sourceFolder url="file://$MODULE_DIR$/spec" isTestSource="true" />
+      <sourceFolder url="file://$MODULE_DIR$/tests" isTestSource="true" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (fritzmg) [inline .idea/docs.iml]: Do we need that in the repo?

File: .idea/.gitignore [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
```
Comment (fritzmg) [inline .idea/.gitignore]: What is this for?

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+* [Global templates](#global-templates) and [Theme specific templates](#theme-specific-templates)
+
+When creating, moving, or renaming Twig templates, we recommend that you use the
+[debug mode](/en/system/debug-mode/#contao-4-8-and-up) to ensure that the newly created templates are used.  
+After completion, the application cache must be rebuilt using the Contao manager or the console.
+
+{{% notice info %}}
+Even in debug mode, it may be necessary to clear the application cache in some cases to make your customized template
+takes effect.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
take effect.
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+title: "Manage templates"
+description: "The management of template files."
+url: "layout/templates/twig/manage"
+weight: 30
+---
+
+## Folder structure
+
+You can store your own templates in the folder `/templates`, so that they are offered for selection in the back end,
+e.g. during the configuration of a content element. Contao distinguishes in the template folder between:
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
e.g. during the configuration of a content element. Within the template folder Contao distinguishes between:
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+{{% notice info %}}
+Even in debug mode, it may be necessary to clear the application cache in some cases to make your customized template
+takes effect.
+{{% /notice %}}
+
+
+## Global templates
+
+Custom global Twig templates are stored inside `/templates` in a special folder structure. The subfolder
+`content_element` is for example for content elements.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
`content_element` is for content elements for example.
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+{{% example "Global template for the text element" %}}
+You want to customize the template for the text element. Select the template `text [content_element/text.html.twig]`
+from the original templates. The necessary structure is given in square brackets after the template name. The template
+is automatically created in the folder `/templates/content_element`.
+{{% /example %}}
+
+These template changes apply to all elements of the respective type. This is not always desired. If you want to
+template changes only for certain elements, different variants of the template are required, which can then be
+selected in the back end of the corresponding element.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
These template changes apply to all elements of the respective type. This is not always desired. If you want to
make changes only for certain elements, different variants of the template are required, which can then be
selected in the back end of the corresponding element.
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+subfolder of the new structure. The name of the variant template has to be defined in compliance with the
+[naming conventions](https://docs.contao.org/dev/framework/templates/creating-templates/#naming-convention).
+but for reasons of clarity it should give an indication of the purpose of the template adaptation.
+
+{{% example "Variant templates for the text element" %}}
+You want to provide multiple variants of the text element.   
+To do this, create a folder called `text` inside `/templates/content_element`. Inside the new folder
+`/templates/content_element/text` you can now create one or more variants of the text element template, e.g.
+`tip.html.twig` and `highlight.html.twig`. In the back end there are now next to the core template
+`text.html.twig` also your two variant templates to choose from.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
`tip.html.twig` and `highlight.html.twig`. The back end now also offers your two variant templates for selection,
next to the core template.
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+{{% notice tip %}}
+You can also globally customize a core template and additionally create variant templates. This makes the possibilities
+for template customization very flexible.
+{{% /notice %}}
+
+
+## Theme specific templates
+
+In the [Theme Manager](../../../theme-manager/manage-themes/) you can link an existing subfolder to a theme.
+with a theme. This is the **theme folder**.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
In the [Theme Manager](../../../theme-manager/manage-themes/) you can link an existing subfolder to a theme.
This is the **theme folder**.
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+
+{{% notice warning %}}
+The name of the theme folder must not be changed for technical reasons
+([managed namespace](https://docs.contao.org/dev/framework/templates/architecture/#managed-namespace)), the name of the
+underscores.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
For technical reasons the folder name must not contain an underscore 
(see [here](https://docs.contao.org/dev/framework/templates/architecture/#managed-namespace)).
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+{{% /notice %}}
+
+Templates in the theme folder are **theme specific templates**. They are special in terms of their handling, because
+although they are the most specific, they are not part of the [template hierarchy](../reuse/#template-hierarchy).
+Only at runtime it is decided whether a theme-specific template is used.
+
+{{% notice note %}}
+You can only use theme specific templates to customize templates that are defined as global templates in the
+templates [template hierarchy](../reuse/#template-hierarchy).
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
You can only use theme specific templates to customize templates that are defined as global templates in the [template hierarchy](../reuse/#template-hierarchy).
```

File: docs/manual/layout/templates/twig/twig-template-manage.en.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+`/templates/themeB/content_element/text.html.twig` to store a customization for the text element in theme B.
+{{% /example %}}
+
+
+### Theme specific variant templates
+
+Theme specific variant templates require a global variant template with the same name.
+
+{{% example "Theme specific variant template for the text element" %}}
+If there is a theme specific template`/templates/content_element/text/highlight.html.twig`, you can additionally use
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-manage.en.md]:
```suggestion
If there is a theme specific template `/templates/content_element/text/highlight.html.twig`, you can additionally use
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+* Templates from Core
+* Templates from an extension
+* Templates from an application
+* Global templates
+* Global variant templates
+
+Templates from the core can be modified by extension and application. For your
+[global templates](../manage/#global-templates) from the `/templates` folder they are parent
+templates that you can customize.   
+The parent templates (base templates) of the core, from extensions or application can be replaced by a
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
The parent templates (base templates) of the core, from extensions or the application can be replaced by a
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+[global-template](../manage/#global-templates) from the folder `/templates` once basically for all elements
+of a type or global [variant-templates](../manage/#global-variants-templates) can be provided for all
+made available. A variant template can either directly be a template from the core from the core, extensions or
+applications or a global template from the folder `/templates`.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
[global-template](../manage/#global-templates) within the `/templates` folder for all elements.
Global [variant templates](../manage/#global-variants-templates) can also be provided. A variant template adjusts a template from the core, an extension or the application - or another global template from the `/templates` folder.
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+
+In addition, [theme-specific templates](../manage/#theme-specific-templates) are available, but they do not
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
In addition, [theme specific templates](../manage/#theme-specific-templates) are available, but they do not
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+belong to the template hierarchy because they are created at runtime.
+
+The template hierarchy can be displayed from the command line with the
+[debug:contao-twig command](https://docs.contao.org/dev/framework/templates/debugging/#debug-contao-twig-command).
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
[`debug:contao-twig` command](https://docs.contao.org/dev/framework/templates/debugging/#debug-contao-twig-command).   
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+of a type or global [variant-templates](../manage/#global-variants-templates) can be provided for all
+made available. A variant template can either directly be a template from the core from the core, extensions or
+applications or a global template from the folder `/templates`.
+
+In addition, [theme-specific templates](../manage/#theme-specific-templates) are available, but they do not
+belong to the template hierarchy because they are created at runtime.
+
+The template hierarchy can be displayed from the command line with the
+[debug:contao-twig command](https://docs.contao.org/dev/framework/templates/debugging/#debug-contao-twig-command).   
+Also note the [corresponding section in the manual](/en/cli) when using the command line command.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
Also note the [corresponding section in the manual](/en/cli) when using the command.
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 20]
```diff
@@ -0,0 +1,178 @@
...
+
+All adjustments must be made within the available blocks.
+
+With `{{ parent() }}` the original content of the block can be output.
+
+All non-customized blocks are automatically taken from the parent template.
+
+{{% example "Extension for the text element" %}}
+We create a new template for the content element text via the back end and insert an additional text into the block
+`{% block text %}` block with an additional text.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
We create a new template for the content element text via the back end and insert an additional text into the `{% block text %}` block with an additional text.
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 21]
```diff
@@ -0,0 +1,178 @@
...
+{% endblock %}
+```
+
+If we now select the template `content_element/text/tip`, then the parent template is the template  
+`/templates/content_element/text.html.twig`.  
+At the beginning again the text `Introductory text for all text elements` is output and additionally at the end the
+text `Here is an additional closing text for the variant "Tip"`. Between these two texts is the complete
+text that we entered in TinyMCE.  
+When using the template `content_element/text/notice`, the closing text `Here is an additional closing text for the
+text for the "Notice" variant`.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
If we now select the template `content_element/text/tip`, then the parent template is the template  
`/templates/content_element/text.html.twig`.  
At the beginning the text `Introductory text for all text elements` is output and additionally at the end the
text `Here is an additional closing text for the variant "Tip"`. Between these the text that we entered in TinyMCE is output.  
When using the template `content_element/text/notice`, the text `Here is an additional closing text for the
text for the "Notice" variant` is output at the end.
```

File: docs/manual/layout/templates/twig/twig-template-reuse.en.md [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+{{% /example %}}
+
+
+### Customize HTML attributes
+
+For most Contao components, you can assign an additional CSS ID or CSS class in the back end.  
+These are set for the corresponding components. Sometimes this is not desired and you want to assign the class
+only for a specific HTML tag.  
+For this purpose Contao provides you with the `{{ attrs() }}` function. With this function, it is possible to set
+as a variable in the parent template.
```
Comment (fritzmg) [inline docs/manual/layout/templates/twig/twig-template-reuse.en.md]:
```suggestion
For most Contao components, you can assign an additional CSS ID or CSS class in the back end.  
These are set for the corresponding components. Sometimes this is not desired and you want to assign the class
only for a specific HTML tag.  
For this purpose Contao provides you with the `{{ attrs() }}` function. This function allows you to modify any HTML attributes that have been defined in the parent template.
```

**Review by fritzmg — CHANGES_REQUESTED** [review]

File: .idea/docs.iml (hunk 4 above)
Comment (netzarbeiter) [reply .idea/docs.iml]: No

File: .idea/.gitignore (hunk 5 above)
Comment (netzarbeiter) [reply .idea/.gitignore]: No

**Review by fritzmg — APPROVED** [review]

**Pull request merged**